gapic/samplegen/samplegen.py

```diff
@@ -332,23 +332,30 @@ def validate_and_transform_request(self,
 
                 if attr.message:
                     base = attr.message
-
+                elif attr.enum:
+                    # A little bit hacky, but 'values' is a list, and this is the easiest
+                    # way to verify that the value is a valid enum variant.
+                    witness = next((e.name
+                                    for e in attr.enum.values
+                                    if e.name == val),
+                                   None)
+                    if not witness:
+                        raise InvalidEnumVariant(
+                            "Invalid variant for enum {}: '{}'".format(attr, val))
+                    # Python code can set protobuf enums from strings.
+                    # This is preferable to adding the necessary import statement
+                    # and requires less munging of the assigned value
+                    duplicate["value"] = f"'{witness}'"
+                    break
                 else:
                     raise TypeError
 
-                # TODO: uncomment this when handling enums
-                # if attr.enum:
-                #     # A little bit hacky, but 'values' is a list, and this is the easiest
-                #     # way to verify that the value is a valid enum variant.
-                #     witness = next((e.name for e in attr.enum.values if e.name == val), None)
-                #     if not witness:
-                #         raise InvalidEnumVariant
-                #     # Python code can set protobuf enums from strings.
-                #     # This is preferable to adding the necessary import statement
-                #     # and requires less munging of the assigned value
-                #     duplicate["value"] = f"'{witness}'"
-
-            # TODO: what if there's more stuff in the chain?
+            if i != len(attr_chain) - 1:
+                extra_attrs = ".".join(attr_chain[i:])
+                raise InvalidEnumVariant(
+                    f"Subsequent attributes in enum request setup: '{extra_attrs}'")
```

Comment:
It's probably just me, but I didn't find this message illuminating. How about: `"Attempted to reference attributes of enum value"`? Also, inside this `if` statement I would add a comment saying `# we broke out of the loop after processing an enum`